src/cls/rbd/cls_rbd_types.h

```diff
@@ -133,6 +133,7 @@ inline void decode(MirrorImageStatusState &state, bufferlist::iterator& it)
 
 struct MirrorImageStatus {
   MirrorImageStatus() {}
+  ~MirrorImageStatus() {}
```

Comment:
Again -- not related to the issue at hand and only is applicable if you deleted a pointer to base (which realistically you would never use). There are a lot of ways to do stupid things w/ C/C++ -- it doesn't make sense to attempt to protect against them all.